### ISSUE TITLE
Documenting Windows support (Cygwin/DKML)

### ISSUE THREAD

**Issue by csaltachin** [aantron/dream#313]
Hey all, I was looking through the README/docs/issues but couldn't find anything too clear about current Windows support. I'm still not very knowledgeable when it comes to OCaml CIs, or building OCaml projects/libraries in general, so I'm happy to be corrected -- but looking at this repo's GitHub Actions test workflow (e.g. latest run [here](https://github.com/aantron/dream/actions/runs/7328813938/job/19957604347)), it seems like the only Windows OCaml/opam environment being targeted is the Cygwin + `opam-repository-mingw` environment ([**"OCaml for Windows"**](https://fdopen.github.io/opam-repository-mingw/)).

There is another major OCaml Windows distribution since at least a year ago (?) called **Diskuv OCaml (DKML)** (OCaml docs [here](https://ocaml.org/docs/ocaml-on-windows), DKML installer repo [here](https://github.com/diskuv/dkml-installer-ocaml#readme)). From my understanding, it uses the same compiler version (4.14.0), but does not overlay the official opam repo with `opam-repository-mingw` by default, does not support `opam-depext`, uses different building tools for more "native" Windows support (?), and so on. (Again, I'm happy to be corrected on the specifics.)

The thing is that, in the ocaml.org docs linked above, DKML appears as a recommendation with better overall support than the Cygwin environment (which is listed as deprecated!). So I think it'd be nice to add some small note on the README regarding which Windows environment(s) is Dream currently intended to support/to be tested on. It's probably not a big deal, but it could help Windows folks interested in Dream have an easier time figuring out whether they'd rather just switch to WSL, or switch to the current Cygwin environment, or wait for tier 1 Windows support with opam 2.2.

Incidentally, here is what prompted me to ask about this: I installed a DKML environment and was following one of the very first tutorials in the OCaml docs ([here](https://ocaml.org/docs/your-first-program#installing-and-using-modules-from-a-package)), where the example is to install Dream via opam and use the library in a project. But `dream` depends indirectly on `conf-pkg-config` and I run into ocaml/opam-repository#24851. And, well, I don't think either of these is necessarily a Dream problem, and WSL is a convenient enough alternative for me, but I can see more people getting tripped up like I did.

**Comment by aantron** [comment]
Indeed, Dream currently aims to support only MinGW, as I haven't had time to try DKML yet.

By the way, I initially started out in OCaml on Windows in Cygwin, but with MSVC. However, this was before opam, and for the last many years, the MinGW approach was much easier.

I think we should add a note to the README. As for what to do, it may be best to wait for opam 2.2 to be ready.

**Comment by yawaramin** [comment]
Just as a data point, I have been building dream-html with GitHub Actions for a while now on all three major platforms, including Windows: https://github.com/yawaramin/dream-html/actions/runs/16553186743/job/46810964485

This is using the 'setup-ocaml' GitHub action and I believe that uses the standard OCaml for Windows installer with whatever Unix shims are needed for a smooth install.

Long story short, I think it will be fairly easy to get the CI running again for Dream.
